exec/java-exec/src/main/codegen/templates/ComparisonFunctions.java

```diff
@@ -215,6 +192,36 @@ public void eval() {
     }
   }
 
+  <#-- IS_DISTINCT_FROM function -->
+  @FunctionTemplate(names = {"is_distinct_from", "is distinct from" },
```

Comment:
I added tests for each category of template code path (primitive type, decimal type and interval type) in TestIsDistinctFromFunctions.java